### DISCUSSION TITLE
How can I inspect machine that is running not in browser?

### DISCUSSION THREAD

**Discussion by azamuddin** [statelyai/xstate#4048] [Q&A] [answered]
I'm trying to use xstate in a scriptkit environment (https://scriptkit.com). 

Since it doesn't have `Window` nor `document` object using `@xstate/inspect` didn't work for me. 

Is it possible to visualize my state machine in such case?

**Comment by chaitanyaj2222** [comment]
@azamuddin , Have you tried this [Stately's online state machine visualizer](https://stately.ai/viz) to visualize your machine?

**Reply by azamuddin** [reply]
You mean copy pasting my machine definition to stately web? 

What I want is the inspect machine using @xstate/inspect so that I can inspect what is going on when I interact with the app.

**Reply by chaitanyaj2222** [reply]
> You mean copy pasting my machine definition to stately web?

Yes. I meant this. 

>What I want is the inspect machine using @xstate/inspect so that I can inspect what is going on when I interact with the app.

I have no clue about this.

**Comment by davidkpiano** [comment, marked as answer]
You can use websockets.

We are currently reworking `@xstate/inspect` and will write updated docs on how to do this.

**Reply by chaitanyaj2222** [reply]
@davidkpiano, I would like to contribute to this. Where can I find the issues related to this? I checked the issues with label 'xstate/inspect' but all of them are stale. If this is not possible, then please redirect me to something that Stately needs help with. Thanks!
